hibernate-core/src/main/java/org/hibernate/graph/internal/AbstractGraph.java

```diff
@@ -123,7 +123,11 @@ public AttributeNodeImplementor<?> addAttributeNode(AttributeNodeImplementor<?>
 	@Override
 	@SuppressWarnings("unchecked")
 	public <AJ> AttributeNodeImplementor<AJ> findAttributeNode(String attributeName) {
-		final PersistentAttribute<? super J, ?> attribute = managedType.findAttributeInSuperTypes( attributeName );
+		PersistentAttribute<? super J, ?> attribute;
+		attribute = managedType.findConcreteGenericAttribute( attributeName );
+		if ( attribute == null ) {
+			attribute = managedType.findAttributeInSuperTypes( attributeName );
+		}
```

Comment:
It would be nice to only call `findConcreteGenericAttribute` when we know the original attribute is generic. Sadly, this information is not available on the `PersistentAttribute` interface so we would have to do an instanceof check and cast to use the `SqmPathSource#isGeneric` method.

Reply:
Done. 
I also applied the same change to `GraphImplementor#findOrCreateAttributeNode`.
I also squashed and force pushed. Looks ok to me.